src/main/java/pl/codewise/internship/MessageQueueComponent.java

```diff
@@ -0,0 +1,74 @@
+package pl.codewise.internship;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.LinkedBlockingDeque;
+import java.util.stream.Collectors;
+
+public final class MessageQueueComponent implements MessageQueue {
```

Comment:
ogólnie jest spoko i wygląda na pierwszy rzut oka, że powinno działać. 
rozejrzyj się za taką strukturą jak circular buffer i spróbuj to rozwiązać używając tego buffera i synchronizować za pomocą locków :) tak bardziej niskopoziomowo rozwiąż to zadanie, żeby poznać struktury, które siedzą pod spodem w kolekcjach synchronizowanych itd.

Reply:
Zrobiłem dodatkową klasę MessageBuffer i tam przy metodach write i get użyłem ReadWriteLock. z tego co rozumiem to teraz nie muszę nigdzie nic synchronizować, bo kiedy jakiś wątek zakluczy write() to wszystkie wątki (niezależnie czy chcą pisać czy odczytywać) czekają, natomiast przy samym odczytywać może wiele wątków naraz.

Z tymi testami wielowątkowymi to przydałaby mi się mała pomoc, podpowiedź, koło ratunkowe?